src/event_generator.hpp

```diff
@@ -11,38 +11,122 @@
 
 namespace arb {
 
+inline
+postsynaptic_spike_event terminal_pse() {
+    return postsynaptic_spike_event{cell_member_type{0,0}, max_time, 0};
+}
+
 // An event_generator generates a sequence of events to be delivered to a cell.
 // The sequence of events is always in ascending order, i.e. each event will be
 // greater than the event that proceded it, where events are ordered by:
 //  - delivery time;
 //  - then target id for events with the same delivery time;
 //  - then weight for events with the same delivery time and target.
-struct event_generator {
```

Comment:
I would maybe add a comment explaining that here we use a type erasure method to avoid inheritance.